app/graphql/mutations/send_update_email_token.rb

```diff
@@ -0,0 +1,50 @@
+module Mutations
+  class SendUpdateEmailToken < ApplicationQuery
+    class NewEmailMustBeUnique < GraphQL::Schema::Validator
+      def validate(_object, context, value)
+        new_email = value[:new_email]
+        current_user = context[:current_user]
+        current_school = context[:current_school]
+
+        user_with_new_email = User.find_by(email: new_email, school: current_school)
+        if user_with_new_email.present? || new_email.blank? || new_email == current_user.email
+          return I18n.t('shared.email_exists_error')
+        end
+      end
+    end
+
+    argument :new_email, String, required: true
+
+    description 'Update a email.'
+
+    field :success, Boolean, null: false
+
+    validates NewEmailMustBeUnique => {}
```

Comment:
A user should be blocked from requesting an email reset for a sane limit of time to avoid abuse of the system. Maybe 2 mins? 🤔